### DISCUSSION TITLE
Debugging in Eclipse

### DISCUSSION THREAD

**Discussion by xarx00** [tchiotludo/akhq#807] [Q&A]
I am able to build the app from command-line, and also from Eclipse using the built-in Gradle plugin. (I imported the sources as an "Existing Gradle Project".) But I am unable to Debug/Run the app from the Eclipse directly.

When I start the app by running org.akhq.App, the command-line used by Eclipse being

```
javaw.exe -Dmicronaut.config.files=E:/akhq/remote.yaml -Dfile.encoding=UTF-8 "@E:\akhq\.temp-AKHQ-classpath-arg-1631001403978.txt" org.akhq.App
```

the server starts, but browser is unable to load the UI. In the (debug) logs it can be seen that the browser tries to load UI from /ui/ui (for `http://localhost:8080/ui`), e.g.:

```
Request GET /ui/ui/static/js/main.3b9d551a.chunk.js
No matching route: GET /ui/ui/static/js/main.3b9d551a.chunk.js
Route match attribute for request (/ui/ui/static/js/main.3b9d551a.chunk.js) not found
Response 404 - GET /ui/ui/static/js/main.3b9d551a.chunk.js
Publishing event: io.micronaut.http.context.event.HttpRequestTerminatedEvent[source=GET /ui/ui/static/js/main.3b9d551a.chunk.js]
```

I can debug the app only by building and running the built JAR:
```
javaw.exe -Dmicronaut.config.files=E:/akhq/remote.yaml -jar "E:\akhq\build\libs\akhq-0.18.0-all.jar" -Dfile.encoding=UTF-8 "@E:\akhq\.temp-AKHQ Jar-classpath-arg-1631003370097.txt" org.akhq.App
```
In this case, the browser is able to find the UI correctly, but having to run the JAR it is rather inconvenient.

Would you know how to start the app (not the JAR) from Eclipse properly so that the browser finds the UI files and I can debug it?
The file `E:\akhq\.temp-AKHQ-classpath-arg-1631001403978.txt` contains
```
-classpath E:\akhq\bin\main;E:\akhq\bin\test;\akhq\bin\default; and used libraries
```

**Comment by tchiotludo** [comment]
Never used eclipse for java project, so I'm not sure how to deal that. 
In my case, I have a local development server in idea with  : 
- `npm install && npm install start` : start the ui 
- `application-dev.yml` in `src/main/resources/` with my local akhq cluster configuration 
- add in `/etc/hosts` `127.0.0.1 akhq`
- `org.akhq.App` has idea application : the backend 
- `http://localhost:3000` to have the ui connected to java api

**Reply by xarx00** [reply]
Eclipse executes the server by using the command-line command I wrote above. When I execute the command myself from the command-line, it doesn't work either. The problem is evidently in the paths or environment variables. I suppose Idea does it the same way. 

Could you please post the exact command Idea uses, and especially the classpath? If I make it work from my command-line, setting the Eclipse will be easy-peasy :-). Thank you.
